tools/report-converter/codechecker_report_converter/cli.py

```diff
@@ -118,15 +118,46 @@ def process_metadata(metadata) -> Tuple[Dict[str, str], Dict[str, str]]:
     return valid_values, invalid_values
 
 
+class CollectFiles(argparse.Action):
+    """
+    This report-converter tool can be given a set of files and directories as
+    positional command line arguments. This action call all files at the given
+    locations: if an argument is a file then that file is collected, if it's a
+    directory then its content is inspected recursively.
+    """
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: str | Sequence[Any] | None,
+        option_string: str | None = None
+    ):
+        # Type of "values" can have any of the indicated types above. But in
+        # argument parser it is given by "nargs='+'", so it will be a list in
+        # this specific case.
+        assert isinstance(values, Sequence)
```

Comment:
The example about defining custom actions in the python documentation show that checking the number of items can be done in the constructor
```
def __init__(self, option_strings, dest, nargs=None, **kwargs):
        if nargs is None:
            raise ValueError("nargs needed")
        super().__init__(option_strings, dest, **kwargs)
```
https://docs.python.org/3/library/argparse.html#action
https://docs.python.org/3/library/argparse.html#argparse.Action

Reply:
Thanks, I didn't notice that there is a standard place for checking this precondition :)